src/util/alloc/bumpallocator.rs

```diff
@@ -32,11 +34,14 @@ pub struct BumpAllocator<VM: VMBinding> {
 #[repr(C)]
 #[derive(Copy, Clone)]
 pub struct BumpPointer {
+    /// The current address of the allocation buffer. This is what will be allocated next.
```

Comment:
I would not say the `cursor` *is* the current address of the allocation buffer itself.  Intuitively, the buffer doesn't move, but the cursor moves.  The buffer may be a whole block or a consecutive range of lines in an Immix block, but it is chosen in the slow path as the range from the *initial* `cursor` to the `limit`.  We don't keep track of the starting point of the buffer, but conceptually the cursor is not the "address of the allocation buffer".

```suggestion
    /// The cursor inside the allocation buffer where the next object will be allocated.
```